packages/flutter_tools/lib/src/commands/analyze_base.dart

```diff
@@ -129,11 +134,12 @@ class PackageDependency {
   }
   bool get hasConflict => values.length > 1;
   bool get hasConflictAffectingFlutterRepo {
-    assert(globals.fs.path.isAbsolute(Cache.flutterRoot));
+    final String? flutterRoot = Cache.flutterRoot;
+    assert(flutterRoot != null && globals.fs.path.isAbsolute(flutterRoot));
```

Comment:
You can get both the "assert not null" _and_ the local promotion using this style:

```suggestion
    flutterRoot!;
    assert(globals.fs.path.isAbsolute(flutterRoot));
```